prow/e2e-suite.sh

```diff
@@ -54,23 +71,25 @@ wget $LINUX_DIST_URL
 tar -xzf ${DAILY_BUILD}-linux.tar.gz
 cp -R ${DAILY_BUILD}/install/* install/
 
-#  Artifact dir is hardcoded in Prow - boostrap to be in first repo checked out
-ARTIFACTS_DIR="${GOPATH}/src/github.com/istio-releases/daily-release/_artifacts"
+get_resource "${RESOURCE_TYPE}" "${OWNER}" "${INFO_PATH}" "${FILE_LOG}"
+setup_cluster
 
 echo 'Running E2E Tests'
-ARGS=( --test_logs_path="${ARTIFACTS_DIR}" --istioctl_url "${ISTIOCTL_STAGE_URL}" "$@" )
-
-EXTRA_E2E_ARGS=()
-EXTRA_E2E_ARGS+=( --mixer_tag="${TAG}" )
-EXTRA_E2E_ARGS+=( --pilot_tag="${TAG}" )
-EXTRA_E2E_ARGS+=( --proxy_tag="${PROXY_TAG:-${TAG}}" )
-EXTRA_E2E_ARGS+=( --ca_tag="${TAG}" )
-EXTRA_E2E_ARGS+=( --mixer_hub="${HUB}" )
-EXTRA_E2E_ARGS+=( --pilot_hub="${HUB}" )
-EXTRA_E2E_ARGS+=( --proxy_hub="${PROXY_HUB:-${HUB}}" )
-EXTRA_E2E_ARGS+=( --ca_hub="${HUB}" )
+E2E_ARGS=(
+  --ca_hub="${HUB}"
+  --ca_tag="${TAG}"
+  --istioctl_url "${ISTIOCTL_STAGE_URL}"
+  --mason_info="${INFO_PATH}"
+  --mixer_hub="${HUB}"
+  --mixer_tag="${TAG}"
+  --pilot_hub="${HUB}"
+  --pilot_tag="${TAG}"
+  --proxy_hub="${PROXY_HUB:-${HUB}}"
+  --proxy_tag="${PROXY_TAG:-${TAG}}"
+  --test_logs_path="${ARTIFACTS_DIR}"
+)
 
 # The --default_proxy flag overwrites both --proxy_hub  and --proxy_tag
-go test -v -timeout 20m ./tests/e2e/tests/simple -args ${ARGS[@]} ${EXTRA_E2E_ARGS[@]}
-go test -v -timeout 20m ./tests/e2e/tests/mixer -args ${ARGS[@]} ${EXTRA_E2E_ARGS[@]}
-go test -v -timeout 40m ./tests/e2e/tests/bookinfo -args ${ARGS[@]} ${EXTRA_E2E_ARGS[@]}
+go test -v -timeout 20m ./tests/e2e/tests/simple -args ${E2E_ARGS[@]} "$@"
```

Comment:
why are we doing this instead of make targets?
this is brittle and will break again

Reply:
I think we have gone through similar discussion and the conclusion was not to rely on makefile from istio/istio since it always changes beneath us. Defining an explicit set of flags to use in testing means we have full control.

Well, except there are cases where we could like the changes in makefile to also carry over to the release tests. By calling go test directly, we will also need to duplicate the flags manually. 

Due to other loads on the three of us, I guess the strategy for now would be to keep this running for a while and switch if it becomes painful. Lets keep the discussion going.